src/main/kotlin/org/rust/ide/RustfmtWatcher.kt

```diff
@@ -0,0 +1,66 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide
+
+import com.intellij.AppTopics
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.components.BaseComponent
+import com.intellij.openapi.editor.Document
+import com.intellij.openapi.fileEditor.FileDocumentManagerListener
+import com.intellij.openapi.project.guessProjectForFile
+import com.intellij.util.containers.ContainerUtil
+import org.rust.cargo.project.settings.rustSettings
+import org.rust.cargo.project.settings.toolchain
+import org.rust.openapiext.virtualFile
+
+class RustfmtWatcher : BaseComponent {
+    private val documentsToReformatLater: MutableSet<Document> = ContainerUtil.newConcurrentSet()
+    @Volatile
+    var isEnabled: Boolean = true
```

Comment:
Why do we need volatile here?
We always use (and looks like must use) it in EDT